addons/mass_mailing/models/mailing.py

```diff
@@ -488,8 +498,19 @@ def action_test(self):
         }
 
     def action_launch(self):
-        self.write({'schedule_type': 'now'})
-        return self.action_put_in_queue()
+        mailing_send_immediate_nbr = self.env['ir.config_parameter'].sudo().get_param('mass_mailing.mass_mailing_immediate_send_nbr', 20)
```

Comment:
I would support that any Falsy value means: don't use immediate send (0, False, '', or config parameter not set). Otherwise you may end up trying to send big mailings ^^ .